Src/Sentry.Xamarin/Extensions/IHubExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Sentry.Extensions
+{
+    internal static class IHubExtensions
+    {
+        internal const string DuplicatedBreadcrumbDropped = "Duplicated breadrumb dropped.";
+
+        /// <summary>
+        /// Adds an automatic breadcrumb to the current scope if the previous one wasn't the same.
+        /// </summary>
+        /// <param name="hub">The Hub which holds the scope stack</param>
+        /// <param name="options">The SentryXamarinOptions that holds the last breadcrumb info.</param>
+        /// <param name="message">The message.</param>
+        /// <param name="category">Category.</param>
+        /// <param name="type">Breadcrumb type.</param>
+        /// <param name="data">Additional data.</param>
+        /// <param name="level">Breadcrumb level.</param>
+        internal static void AddInternalBreadcrumb(this IHub hub, SentryXamarinOptions options, string message, string? category = null, string? type = null, Dictionary<string, string>? data = null, BreadcrumbLevel level = BreadcrumbLevel.Info)
+        {
+            //Filter duplicated internal breadcrumbs
+            if (options.LastInternalBreadcrumb != null &&
```

Comment:
This is a static method and will be called by multiple threads.
Each of these `options.Property` calls could result in reading a new instance. So either you synchronize things (`lock` which could result in contention though no I/O is done here so probably OK (measuring is the way to know) or you take a copy of the reference to a local variable and do the checks with` localvar.Message && localvar.Category`. The downside then is that you could be checking on a crumb that's no longer relevant since a new one was already added to `LastInternalBreadcrumb`. It's probably good enough for a dedupe strategy though.

Reply:
Good observation, I like the second approach more.